backends/magma/weight_1d.cu

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2017-2018, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory. LLNL-CODE-734707.
+// All Rights reserved. See files LICENSE and NOTICE for details.
+//
+// This file is part of CEED, a collection of benchmarks, miniapps, software
+// libraries and APIs for efficient high-order finite element and spectral
+// element discretizations for exascale applications. For more information and
+// source code availability see http://github.com/ceed.
+//
+// The CEED research is supported by the Exascale Computing Project 17-SC-20-SC,
+// a collaborative effort of two U.S. Department of Energy organizations (Office
+// of Science and the National Nuclear Security Administration) responsible for
+// the planning and preparation of a capable exascale ecosystem, including
+// software, applications, hardware, advanced system engineering and early
+// testbed platforms, in support of the nation's exascale computing imperative.
+
+#include <ceed.h>
+#include <cuda.h>    // for CUDA_VERSION
+#include <magma_v2.h>
+#include "magma_common_device.cuh"
+#include "weight_device.cuh"
+
+//////////////////////////////////////////////////////////////////////////////////////////
+extern __shared__ CeedScalar shared_data[];
+template<typename T, int Q>
+static __global__ void
+magma_weight_1d_kernel(const T *dqweight1d, T *dV, const int v_stride, const int nelem)
+{
+    const int tx      = threadIdx.x;
+    const int ty      = threadIdx.y;
+    const int elem_id = (blockIdx.x * blockDim.y) + ty;
+
+    if(elem_id >= nelem) return;
+
+    // global memory pointers
+    dV += elem_id * v_stride;
+
+    // shared memory pointers
+    T* sTweight = (T*)shared_data;
+    T* sV = sTweight + Q;
+    sV   += ty * Q;
+
+    // read dqweight_1d
+    if(ty == 0 && tx < Q) {
+        sTweight[tx] = dqweight1d[tx];
+    }
+
+    __syncthreads();
+    magma_weight_1d_device<T, Q>(sTweight, sV, tx);
+    __syncthreads();
+
+    // write V
+    dV[ tx ] = sV[ tx ];
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////
+template<typename T, int Q>
+static magma_int_t 
+magma_weight_1d_kernel_driver(
+    const T *dqweight1d, T *dV, magma_int_t v_stride, 
+    magma_int_t nelem, magma_int_t maxthreads, magma_queue_t queue)
+{
+    magma_device_t device;
+    magma_getdevice( &device );
+    magma_int_t shmem_max, nthreads_max;
+
+    magma_int_t nthreads = Q; 
+    magma_int_t ntcol = (maxthreads < nthreads) ? 1 : (maxthreads / nthreads);
+    magma_int_t shmem  = 0;
+    shmem += sizeof(T) * Q;  // for dqweight1d 
+    shmem += sizeof(T) * ntcol * Q; // for output
+
+    cudaDeviceGetAttribute (&nthreads_max, cudaDevAttrMaxThreadsPerBlock, device);
+    #if CUDA_VERSION >= 9000
+    cudaDeviceGetAttribute (&shmem_max, cudaDevAttrMaxSharedMemoryPerBlockOptin, device);
+    if(shmem <= shmem_max) {
```

Comment:
I don't know how much this is a style point we care about, but almost exclusively we use spaces after `if` and `case` throughout the library.

Reply:
I think I got all the `if`s in 450c88733e840de4f5386a5529f50eb0e4326511